src/wallet/interfaces.cpp

```diff
@@ -177,11 +177,7 @@ class WalletImpl : public Wallet
     }
     bool haveWatchOnly() override
     {
-        auto spk_man = m_wallet->GetLegacyScriptPubKeyMan();
-        if (spk_man) {
-            return spk_man->HaveWatchOnly();
-        }
-        return false;
+        return m_wallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS);
```

Comment:
In dc7bf5fd6a320c4528a28cef2a565366bbab3877 "wallet: Delete LegacySPKM": why doesn't this just return `false` like before? And if so, can't the entire helper be dropped?

Reply:
Indeed, removed.